src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/WorkspaceProjectContextProvider.cs

```diff
@@ -66,8 +65,9 @@ public async Task ReleaseProjectContextAsync(IWorkspaceProjectContextAccessor ac
             {
                 accessor.Context.Dispose();
             }
-            catch (Exception ex) when (_telemetryService.PostFault(TelemetryEventName.LanguageServiceInitFault, ex))
```

Comment:
Are there any situations where `PostFault` would return `false` and we would let the exception escape?

Really what I'm asking is: are we sure we're preserving the same behavior around which exceptions we catch and which escape?

Reply:
No, it always return true.

However, there is a subtle behavior change here; before we'd PostFault before stack was unwound, which meant if we got a dump we'd have no run any finallys at this point. The thing is though, PostFault doesn't actually ever create a dump - so that was pointless anyway.

I want to way to come up with a way that ProjectFaultHandlerService can be used in an exception filter, but it requires thought because it's all async all the way down.